backend/src/chains/multi_retriever_chain.py

```diff
@@ -0,0 +1,134 @@
+        prompt_template_str: Optional[str] = None,
+        docs_path: Optional[list[str]] = None,
+        manpages_path: Optional[list[str]] = None,
+        embeddings_model_name: Optional[str] = None,
+        use_cuda: bool = False,
+        search_k: list[int] = [5, 5],
+        weights: list[float] = [0.5, 0.5],
+        chunk_size: int = 500,
+    ):
+        super().__init__(
+            llm_model=llm_model,
+            prompt_template_str=prompt_template_str,
+        )
+        self.embeddings_model_name: Optional[str] = embeddings_model_name
+        self.use_cuda: bool = use_cuda
+
+        self.search_k: list[int] = search_k
+        self.weights: list[float] = weights
+
+        self.chunk_size: int = chunk_size
+        self.docs_path: Optional[list[str]] = docs_path
+        self.manpages_path: Optional[list[str]] = manpages_path
+
+        self.retriever: EnsembleRetriever = None
+
+    def create_multi_retriever(
+        self,
+    ) -> EnsembleRetriever:
+        docs_similarity_retriever_chain = SimilarityRetrieverChain(
+            llm_model=None,
+            prompt_template_str=None,
+            embeddings_model_name=self.embeddings_model_name,
+            docs_path=self.docs_path,
+            chunk_size=self.chunk_size,
+        )
+        docs_similarity_retriever_chain.embed_docs(return_docs=False)
+        docs_similarity_retriever_chain.create_similarity_retriever(search_k=5)
+        docs_similarity_retriever = docs_similarity_retriever_chain.retriever
+
+        manpages_similarity_retriever_chain = SimilarityRetrieverChain(
+            llm_model=None,
+            prompt_template_str=None,
+            embeddings_model_name=self.embeddings_model_name,
+            manpages_path=self.manpages_path,
+            chunk_size=self.chunk_size,
+        )
+        manpages_similarity_retriever_chain.embed_docs(return_docs=False)
+        manpages_similarity_retriever_chain.create_similarity_retriever(search_k=5)
+        manpages_similarity_retriever = docs_similarity_retriever_chain.retriever
+
+        if (
+            docs_similarity_retriever is not None
+            and manpages_similarity_retriever is not None
+        ):
+            self.retriever = EnsembleRetriever(
+                retrievers=[docs_similarity_retriever, manpages_similarity_retriever],
+                weights=self.weights,
+            )
+
+    def create_llm_chain(self) -> None:
+        super().create_llm_chain()
+
+        self.llm_chain = (
+            RunnablePassthrough.assign(context=(lambda x: format_docs(x["context"])))
+            | self.llm_chain
+        )
+
+        llm_chain_with_source = RunnableParallel({
+            "context": self.retriever,
+            "question": RunnablePassthrough(),
+        }).assign(answer=self.llm_chain)
+
+        self.llm_chain = llm_chain_with_source
+
+        return
+
+
+if __name__ == "__main__":
+    load_dotenv()
+
+    # from langchain_google_vertexai import ChatVertexAI
```

Comment:
Add TODO for next PR: There should be logic that handles credentials for both vertex ai and ai studio. This should not be a comment.